test/regression/issue/11806.test.ts

```diff
@@ -0,0 +1,38 @@
+import { test, expect } from "bun:test";
+import { bunExe, tempDirWithFiles } from "harness";
+
+test("11806", () => {
+  const dir = tempDirWithFiles("11806", {
+    "package.json": JSON.stringify({
+      "name": "project",
+      "workspaces": ["apps/*"],
+    }),
+    "apps": {
+      "api": {
+        "package.json": JSON.stringify({
+          "name": "api",
+          "jest": {
+            "testRegex": ".*\\.spec\\.ts$",
+          },
+          "devDependencies": {
+            "typescript": "^5.7.3",
+          },
+        }),
+      },
+    },
+  });
+
+  const result1 = Bun.spawnSync({
+    cmd: [bunExe(), "install"],
+    stdio: ["inherit", "inherit", "inherit"],
+    cwd: dir + "/apps/api",
+  });
+  expect(result1.exitCode).toBe(0);
+
+  const result2 = Bun.spawnSync({
```

Comment:
Consider running these in parallel (e.g. `await Promise.all([ Bun.spawn({ ... }), ... ])`